### PR TITLE
Player Self-Registration

### PR DESCRIPTION
**Pull request by armerpaul**
## Before Testing!
- Start up your app as normal
- Run the database migrations: `docker-compose run app rake db:migrate 
`

### Set up self-registration
- create a tournament, checking the self-registration box
image

- verify you see the "Register for this Event" box
image

- navigate to the "Settings" tab, verify you see the self-registration box is checked
image

### TO registers as a player
- create a tournament with self-registration
- In the  "Register for this Event" box:
    - Verify your NRDB username is prefilled
    - Enter a runner and corp ID
image

- Click the button, then verify you see your registration information
image

### Regular player self-registers
- log out and navigate to a tournament with self-registration
- verify you see a yellow "Register for this Event" box
image

- log in, then navigate back to the tournament (future TODO: redirect to the tournament)
- verify you see the self-registration form
image

- fill out the form and verify you see your information
image
 
### Dropped Player Message
- Register for a tournament with an account that isn't the TO
- Log in as the TO and drop that player
- Verify they see this message
image

### Rejoin tournament as TO
- Create a tournament, and self-register for the tournament
- Drop yourself in the Players tab
- Navigate back to the Tournament tab, verify you see this message: 
image

### Players can always see Registration Info
- As a TO, self-registrater for one of your tournaments
- Navigate to the settings tab and disable self-registration
- Navigate back to the Tournament tab, verify you still see your registration information
image

### Players Tab UI Changes
- The players tab should have all original functionality, I just changed up the UI a bit
image

### CODE REVIEW

File: app/assets/stylesheets/global.sass [hunk 1]
```diff
@@ -22,19 +22,9 @@ a
     text-decoration: none
 
 nav.navbar, .footer
-  background: rgb(37, 42, 58)
```
Comment (plural) [inline app/assets/stylesheets/global.sass:25]: why these changes?  I noticed that the footer is a different shade now than it was before.

File: app/assets/stylesheets/players.sass [hunk 2]
```diff
@@ -1,3 +1,19 @@
...
+    margin-bottom: 0.75rem
+
+.form-inline
+  .identities_form
+    > *
+      display: inline-block
+      margin-bottom: 0
+
+      &:not(:last-child)
+        margin-right: 0.5rem
```
Comment (plural) [inline app/assets/stylesheets/players.sass:19]: add the trailing newline to prevent warnings later.

File: app/views/layouts/application.html.slim [hunk 3]
```diff
@@ -68,12 +74,13 @@ html
...
-    .footer.dontprint
+    .footer.dontprint.bg-dark.text-light
       .container
         .row
-          .col-4.nav-item
-            => link_to 'Support NISEI on Patreon', 'https://www.patreon.com/join/NISEI'
-          .col-4.nav-item.text-center
-           | 🐍 Maintained by NISEI. Created by Johno. 🐍 
-          .col-4.nav-item.text-right
+          .col-3
```
Comment (plural) [inline app/views/layouts/application.html.slim:80]: why these changes?  Typically, things have been in a bootstrap 12-column width, but this shifts this up.

The layout shifted on the bottom navbar:
old:
Screenshot 2022-02-05 12 48 09 PM

new:
Screenshot 2022-02-05 12 44 38 PM

File: app/views/players/_form.html.slim [hunk 4]
```diff
@@ -0,0 +1,33 @@
+
```
Comment (plural) [inline app/views/players/_form.html.slim:1]: drop the top 2 blank lines?

File: app/assets/stylesheets/players.sass [hunk 5]
```diff
@@ -1,3 +1,19 @@
 .players
   .manual_seed
     width: 80px
+
+.identities_form
+  .awesomplete
+    display: block
+  > *
```
Comment (plural) [inline app/assets/stylesheets/players.sass:8]: i don't know sass so i will trust you about the formatting here.  :D The indentation and the > * looks a bit inconsistent, but you tell me if that is fine or not.

**Review by plural — CHANGES_REQUESTED** [review]
I patched this in and it works really well and looks generally great.

A couple of things to tidy up/clarify, but that's it.

**Comment by armerpaul** [comment]
sorry for not seeing these comments @plural! I think I need the github mobile app to give me notifications when I get comments/merges.

File: app/views/layouts/application.html.slim (hunk 3 above)
Comment (armerpaul) [reply app/views/layouts/application.html.slim:80]: I was trying to clean up the navbar and footer as I was poking around to use minimal custom classes. Then I got carried away... 😅 

I'll pull these changes out and save them for a bigger UI cobra improvement project!

File: app/assets/stylesheets/players.sass (hunk 5 above)
Comment (armerpaul) [reply app/assets/stylesheets/players.sass:8]: I was being lazy. `> *` will select all the immediate children of `.identities_form`. I'm changing it to select the two different input containers so it's more clear.

File: app/assets/stylesheets/global.sass (hunk 1 above)
Comment (armerpaul) [reply app/assets/stylesheets/global.sass:25]: I originally wanted to just use the `.bg-dark` class from bootstrap. But we really should do bigger style changes in a more cohesive project, so I changed it back.

**Review by plural — APPROVED** [review]
thanks!
